models/virtual_service.go

```diff
@@ -67,3 +67,129 @@ func (vService *VirtualService) IsValidHost(namespace string, serviceName string
 
 	return kubernetes.FilterByRoute(protocols, protocolNames, serviceName, namespace, nil)
 }
+
+// HasRequestTimeout determines if the spec has an http timeout set.
+func (vService *VirtualService) HasRequestTimeout() bool {
+	if vService == nil {
+		return false
+	}
+
+	if routes, isSlice := vService.Spec.Http.([]interface{}); isSlice {
+		for _, route := range routes {
+			if routeMap, isMap := route.(map[string]interface{}); isMap {
+				if _, hasTimeout := routeMap["timeout"]; hasTimeout {
+					return true
+				}
+			}
+		}
+	}
```

Comment:
imo ideally the `VirtualService` model would be fully typed and then these methods just performing type checking would no longer be necessary. I'm happy to make that change and remove these methods. Could Kiali simply import and use the upstream istio api types e.g. something like `networkingv1beta1.VirtualService`?

Reply:
This is probably a better question for @lucasponce .  I think perhaps the models are parsed out the way they in an effort to simplify the structure to something more easy for us to deal with at higher levels.  Perhaps this badging stuff could be parsed out immediately, or stored lazily on the model, to possibly avoid doing the work many times.

Reply:
@nrfox Now that @lucasponce is on PTO, I think we should leave this alone until he comes back and can comment.

Reply:
I will test this as-is.